air-script/tests/main.rs

```diff
@@ -1,191 +1,327 @@
 use expect_test::expect_file;
 
 mod helpers;
-use helpers::Test;
+use helpers::{Target, Test};
 
 // TESTS
 // ================================================================================================
 
 #[test]
 fn aux_trace() {
-    let generated_air = Test::new("tests/aux_trace/aux_trace.air".to_string())
-        .transpile()
+    let generated_winterfell = Test::new("tests/aux_trace/aux_trace.air".to_string())
+        .transpile(Target::Winterfell)
         .unwrap();
 
     let expected = expect_file!["aux_trace/aux_trace.rs"];
-    expected.assert_eq(&generated_air);
+    expected.assert_eq(&generated_winterfell);
+
+    let generated_masm = Test::new("tests/aux_trace/aux_trace.air".to_string())
+        .transpile(Target::Masm)
+        .unwrap();
+
+    let expected = expect_file!["aux_trace/aux_trace.masm"];
+    expected.assert_eq(&generated_masm);
```

Comment:
I'm not convinced that we should have both of these in the same test (and similar for the rest of the tests in this file). It implies that if we added more backends (e.g. the IR we'll share with PIL or something targeting another zkVM) we would test them all here too. I would probably advocate a different test module for each backend. I assume you considered this though, so it'd be good to hear your thinking

Reply:
I haven't thought much about it, will add a new module 👍

Reply:
I split the tests into 2 separate modules, because each module is a different binary (these are integration tests), the compiler was complaining the other option was not being used (i.e. the winterfell tests didn't generate masm code, so the `Target::Masm` was never used on that compilation unit).

To make the compiler happy, I made `main.rs` import both `winterfell` and `masm`, so all variants of the enum are used, and no warning is emitted. To avoid having the compilation for both tests being done twice, I moved the code into a module named `codegen`.

To summarize:

- `codegen/winterfell.rs` run the winterfell codegen tests
- `codegen/masm.rs` runs the masm
- all tests are inside `codegen` to satisfy the compiler